### ISSUE TITLE
Done anything with 2.15 yet? (Advanced settings)

### ISSUE THREAD

**Issue by awhileback** [coderedcorp/coderedcms#465]
Hey Vince and co, hope you all are doing well.

I don't use your whole distribution of wagtail but I do like your idea of implementing a struct block on all the stream field blocks with the "advanced settings" button allowing CSS customization, so have implemented it in my current project.

On 2.15 it seems to break with your latest patch to that functionality, have you all messed with the 2.15 admin js/css yet?  I just included the script for the functionality in the block form template at the bottom in a script tag, I don't have it included from a separate js file, but I don't think that should matter.

Here's a short video showing the broken behavior...

https://user-images.githubusercontent.com/84097090/150472711-aad9c276-49f3-4f75-9af6-844e2c71d937.mov

**Comment by awhileback** [comment]
Fixed using a callback instead of chaining the functions, which doesn't seem to work.

```
$(document).ready(function(){
    $(document).on('click', '.website-collapsible button', function() {
        var $target = $(this).parent().find('.website-collapsible-target');

        if ($(this).parent().hasClass('collapsed')) {
            $target.show('slow', function() {
                $(this).parent().removeClass('collapsed');
            });
        } else {
            $target.hide('slow', function() {
                $(this).parent().addClass('collapsed');
                
            });
        }
    });
});
```

**Comment by vsalvino** [comment]
Thanks for testing and providing this workaround. We have not implemented 2.15 yet simply because there are a lot of breaking changes and it is proving to be a heavy lift.

For this particular snippet of code, I would recommend to re-implement in plain JS - just because I know the wagtail admin is rapidly changing and plain JS is much more of a sure thing that won't interfere with the admin changes.

**Comment by awhileback** [comment]
Yeah, I noticed that they plan to deprecate jQuery so I thought the same thing, project for another day I suppose.
